server/paymentProviders/stripe/virtual-cards.js

```diff
@@ -217,6 +217,22 @@ export const processDeclinedAuthorization = async (stripeAuthorization, stripeEv
     ? stripeAuthorization.metadata.oc_decline_code
     : stripeAuthorization.request_history[0].reason;
 
+  const expense = await models.Expense.findOne({
+    where: {
+      VirtualCardId: virtualCard.id,
+      data: { authorizationId: stripeAuthorization.id },
+    },
+  });
+
+  if (!expense) {
+    logger.error(
+      `Stripe: could not find expense attached to reversed authorization ${stripeAuthorization.id}`,
+      stripeEvent,
+    );
+    return;
+  }
+  await expense.update({ status: ExpenseStatus.CANCELED });
+
```

Comment:
Could you add a comment and explain when this can be happening from the user standpoint?